lib/src/create_report.dart

```diff
@@ -561,21 +560,52 @@ extension on SourceSpan {
   }
 }
 
+class _Subsection {
+  final List<_Issue> issues;
+  final String description;
+  final int points;
+
+  final _Status status;
+
+  _Subsection(
+    this.description,
+    this.issues,
+    this.points,
+    this.status,
+  );
+}
+
 /// Given an introduction and a list of issues, formats the summary of a
 /// section.
-String _makeSummary(String introduction, List<_Issue> issues,
-    {@required String basePath, int maxIssues = 2}) {
-  final issuesMarkdown = issues.map((e) => e.markdown(basePath: basePath));
+String _makeSummary(List<_Subsection> subsections,
+    {String introduction, @required String basePath, int maxIssues = 2}) {
   return [
-    introduction,
-    if (issues.isNotEmpty) ...[
-      '',
-      if (issues.length <= maxIssues)
-        ...issuesMarkdown
-      else ...[
-        'Found ${issues.length} issues. Showing the first $maxIssues:',
-        ...issuesMarkdown.take(maxIssues),
-      ],
-    ],
+    if (introduction != null) introduction,
+    ...subsections.expand((subsection) {
+      final issuesMarkdown =
+          subsection.issues.map((e) => e.markdown(basePath: basePath));
+      final statusMarker = _statusMarker(subsection.status);
+      return [
+        '### $statusMarker **${subsection.points} points**: ${subsection.description}',
```

Comment:
Do we anticipate different formatting for the points here? `<h3>`? If not, we could drop the `**`s.